utils/spdx/src/test/kotlin/SpdxCompoundExpressionTest.kt

```diff
@@ -93,5 +93,12 @@ class SpdxCompoundExpressionTest : WordSpec({
             // Compare string representations to not rely on semantic equality.
             expression.simplify().toString() shouldBe "MIT"
         }
+
+        "remove redundant choices" {
+            val expression = "Apache-2.0 AND (Apache-2.0 OR MIT) AND MIT".toSpdx()
+
+            // Compare string representations to not rely on semantic equality.
```

Comment:
Should this say "structural equality" instead? I'm actually not sure if I understand the comment.

Reply:
As this is about simplifying expressions, I want to be sure that e.g. `MIT AND MIT` is simplified to exactly `MIT` although it is `"MIT AND MIT".toSpdx() == "MIT".toSpdx()`.

Reply:
Yes, but isn't "MIT AND MIT" semantically equal with "MIT", but not syntactically?

Reply:
Yes, exactly. The comment is meaning to say: "If we would not use string-comparison here, expressions would be compared for being semantically equal, which is not what we want."